File: 1-DuplicateInteger/solution.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public boolean hasDuplicate(int[] nums) {
        // Criamos um HashSet para armazenar os números únicos.
        // O HashSet não permite duplicatas, o que facilita a verificação.
        Set<Integer> seen = new HashSet<>();
        // Percorremos o array nums, iterando sobre cada número.
        for (int num : nums) {
            // Verificamos se o número atual já está no HashSet.
            // Se estiver, significa que encontramos um duplicado e retornamos true.
            if (seen.contains(num)) {
                return true; // Duplicata encontrada
            }
            // Se o número não estiver no HashSet, adicionamos ele.
            seen.add(num);
        }
        
        // Se terminarmos o loop sem encontrar duplicatas, retornamos false.
        return false; // Nenhuma duplicata foi encontrada
    }
}

File: 1-DuplicateInteger/solution.go
package main

import "fmt"

func hasDuplicate(nums []int) bool {
    // Criamos um mapa para armazenar números únicos.
    seen := make(map[int]bool)
    
    // Iteramos sobre o array nums.
    for _, num := range nums {
        // Verificamos se o número já existe no mapa.
        if _, exists := seen[num]; exists {
            return true // Encontramos um número duplicado.
        }
        // Caso contrário, adicionamos o número ao mapa.
        seen[num] = true
    }
    
    // Se chegarmos ao final do loop, não há duplicatas.
    return false
}

func main() {
    nums1 := []int{1, 2, 3, 3} // Exemplo com duplicata
    nums2 := []int{1, 2, 3, 4} // Exemplo sem duplicata
    
    // Testamos a função e imprimimos os resultados
    fmt.Println(hasDuplicate(nums1)) // true
    fmt.Println(hasDuplicate(nums2)) // false
}
